File: src/Test/java/com/nishant/algorithms/Sorting/InsertionSortTest.java
package com.nishant.algorithms.sorting;

import com.nishant.algorithms.sorting.insertionsort.InsertionSort;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;

public class InsertionSortTest {

  private int[] arr;

  @Before
  public void setup() {
    arr = new int[] {5, 3, 2, 1, 5};
  }

  @Test
  public void test() {
    InsertionSort.sort(arr);
    assertArrayEquals(new int[] {1, 2, 3, 5, 5}, arr);
  }
}

File: cpp/test/sorting/insertionsort/insertion_sort_test.cpp
//
// Created by nisha on 6/2/2020.
//

#include <gtest/gtest.h>

#include "../../../main/sorting/insertionsort/insertion_sort.h"

TEST(Sorting, InsertionSortTest) {
  std::vector<int> arr = my_utils::createRandVector(10000, 1, 1000000);
  std::vector<int> copy = arr;
  
  std::sort(copy.begin(), copy.end());
  insertion_sort(arr);
  ASSERT_EQ(copy, arr);
}
